engines/advancedDetector.cpp

```diff
@@ -505,57 +540,85 @@ namespace Common {
 }
 
 // Sync with engines/game.cpp
-static char flagsToMD5Prefix(uint32 flags) {
-	if (flags & ADGF_MACRESFORK) {
-		if (flags & ADGF_TAILMD5)
+static char flagsToMD5Prefix(FilePropertiesFlags flags) {
+	if (flags & kUseMacForks) {
+		if (flags & kUseTail)
 			return 'e';
 		return 'm';
 	}
-	if (flags & ADGF_TAILMD5)
+	if (flags & kUseTail)
 		return 't';
 
 	return 'f';
 }
 
-static bool getFilePropertiesIntern(uint md5Bytes, const AdvancedMetaEngine::FileMap &allFiles, const ADGameDescription &game, const Common::String &fname, FileProperties &fileProps);
+static bool getFilePropertiesIntern(uint md5Bytes, const AdvancedMetaEngine::FileMap &allFiles, FilePropertiesFlags flags, const Common::String &fname, FileProperties &fileProps);
+
+bool AdvancedMetaEngineDetection::getFileProperties(const FileMap &allFiles, FilePropertiesFlags flags, const Common::String &fname, FileProperties &fileProps) const {
+	Common::String hashname = Common::String::format("%c:%s:%d", flagsToMD5Prefix(flags), fname.c_str(), _md5Bytes);
 
-bool AdvancedMetaEngineDetection::getFileProperties(const FileMap &allFiles, const ADGameDescription &game, const Common::String &fname, FileProperties &fileProps) const {
-	Common::String hashname = Common::String::format("%c:%s:%d", flagsToMD5Prefix(game.flags), fname.c_str(), _md5Bytes);
+	// It's tempting to merge datafork: one with prefixless but in case of extensionless macbinary plain file and datafork have different checksums
 
-	if (MD5Man.contains(hashname)) {
+	if ((flags & kUseMacForks) && MD5Man.contains("resfork:" + hashname) && MD5Man.contains("datafork:" + hashname) && MD5Man.contains(hashname)) {
+		fileProps.md5 = MD5Man.getMD5("datafork:" + hashname);
```

Comment:
What about combination of "tail" and res or datafork? This is why I was using the single letters, so I could combine

Reply:
It can combine. It is datafork:t:md5